providers/collection_package.rb

```diff
@@ -92,3 +119,129 @@ def create_collection_package
 
   Opennms::Helpers.write_xml_file(doc, "#{node['opennms']['conf']['home']}/etc/collectd-configuration.xml")
 end
+
+def update_collection_package
+	Chef::Log.debug "Updating collectd-configuration.xml definition : '#{new_resource.name}'"
+	file = ::File.new("#{node['opennms']['conf']['home']}/etc/collectd-configuration.xml")
+	contents = file.read
+	doc = REXML::Document.new(contents, respect_whitespace: :all)
+	doc.context[:attribute_quote] = :quote
+	file.close
+	
+	package_el = matching_package_name(doc, new_resource)
+
+	# put the new ones in
+	unless new_resource.filter.nil?
+		new_resource.filter.each do |filter|
+		if package_el.elements["filter[text() = '#{filter}']"].nil?
+			filter_el = package_el.add_element 'filter'
+			filter_el.add_text(filter)
+			end
+		end
+	end
```

Comment:
filters are a scalar String, not an array. And you need to replace the existing filter. And it's required, you don't need to check if it is nil.